// mailersend/lib/modules/sms/Inbound.module.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmsInboundModule = void 0;
const request_service_1 = require("../../services/request.service");
class SmsInboundModule extends request_service_1.RequestService {
    constructor(apiKey, baseUrl) {
        super(apiKey, baseUrl);
    }
    create(smsInbound) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.post(`/sms-inbounds`, smsInbound);
        });
    }
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/sms-inbounds`, queryParams);
        });
    }
    single(smsInboundId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/sms-inbounds/${smsInboundId}`);
        });
    }
    delete(smsInboundId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.deleteReq(`/sms-inbounds/${smsInboundId}`);
        });
    }
    update(smsInboundId, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.put(`/sms-inbounds/${smsInboundId}`, data);
        });
    }
}
exports.SmsInboundModule = SmsInboundModule;
