// mailersend/lib/modules/sms/Webhook.module.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmsWebhookModule = void 0;
const request_service_1 = require("../../services/request.service");
class SmsWebhookModule extends request_service_1.RequestService {
    constructor(apiKey, baseUrl) {
        super(apiKey, baseUrl);
    }
    create(params) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.post("/sms-webhooks", params);
        });
    }
    list(smsNumberId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get("/sms-webhooks", { sms_number_id: smsNumberId });
        });
    }
    single(smsWebhookId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/sms-webhooks/${smsWebhookId}`);
        });
    }
    update(smsWebhookId, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.put(`/sms-webhooks/${smsWebhookId}`, data);
        });
    }
    delete(smsWebhookId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.deleteReq(`/sms-webhooks/${smsWebhookId}`);
        });
    }
}
exports.SmsWebhookModule = SmsWebhookModule;
