// mailersend/lib/models/sms/Inbound.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmsComparerType = exports.SmsInbound = void 0;
class SmsInbound {
    constructor(name, smsNumberId, forwardUrl, enabled, filter) {
        this.name = name;
        this.enabled = enabled;
        this.sms_number_id = smsNumberId;
        this.forward_url = forwardUrl;
        this.filter = filter;
    }
    setSmsNumberId(smsNumberId) {
        this.sms_number_id = smsNumberId;
        return this;
    }
    setName(name) {
        this.name = name;
        return this;
    }
    setEnabled(enabled) {
        this.enabled = enabled;
        return this;
    }
    setForwardUrl(forward_url) {
        this.forward_url = forward_url;
        return this;
    }
    setFilter(filter) {
        this.filter = filter;
        return this;
    }
}
exports.SmsInbound = SmsInbound;
var SmsComparerType;
(function (SmsComparerType) {
    SmsComparerType["EQUAL"] = "equal";
    SmsComparerType["NOT_EQUQL"] = "not-equal";
    SmsComparerType["CONTAINS"] = "contains";
    SmsComparerType["NOT_CONTAINS"] = "not-contains";
    SmsComparerType["STARTS_WITH"] = "starts-with";
    SmsComparerType["ENDS_WITH"] = "ends-with";
    SmsComparerType["NOT_STARTS_WITH"] = "not-starts-with";
    SmsComparerType["NOT_ENDS_WITH"] = "not-ends-with";
})(SmsComparerType || (exports.SmsComparerType = SmsComparerType = {}));
