// mailersend/lib/models/sms/SMSParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SMSParams = void 0;
class SMSParams {
    constructor(config) {
        var _a;
        this.from = config === null || config === void 0 ? void 0 : config.from;
        this.to = config === null || config === void 0 ? void 0 : config.to;
        this.text = config === null || config === void 0 ? void 0 : config.text;
        if ((_a = config === null || config === void 0 ? void 0 : config.personalization) === null || _a === void 0 ? void 0 : _a.length) {
            this.personalization = config === null || config === void 0 ? void 0 : config.personalization;
        }
    }
    setFrom(from) {
        this.from = from;
        return this;
    }
    setTo(to) {
        this.to = to;
        return this;
    }
    setText(text) {
        this.text = text;
        return this;
    }
    setPersonalization(personalization) {
        this.personalization = personalization;
        return this;
    }
}
exports.SMSParams = SMSParams;
