// mailersend/lib/models/sms/Webhook.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmsWebhookEventType = exports.SmsWebhook = void 0;
class SmsWebhook {
    constructor(name, url, events, smsNumberId, enabled) {
        this.url = url;
        this.name = name;
        this.events = events;
        this.sms_number_id = smsNumberId;
        this.enabled = enabled;
    }
    setUrl(url) {
        this.url = url;
        return this;
    }
    setName(name) {
        this.name = name;
        return this;
    }
    setEvents(events) {
        this.events = events;
        return this;
    }
    setSmsNumberId(smsNumberId) {
        this.sms_number_id = smsNumberId;
        return this;
    }
    setEnabled(enabled) {
        this.enabled = enabled;
        return this;
    }
}
exports.SmsWebhook = SmsWebhook;
var SmsWebhookEventType;
(function (SmsWebhookEventType) {
    SmsWebhookEventType["SENT"] = "sms.sent";
    SmsWebhookEventType["DELIVERED"] = "sms.delivered";
    SmsWebhookEventType["FAILED"] = "sms.failed";
})(SmsWebhookEventType || (exports.SmsWebhookEventType = SmsWebhookEventType = {}));
