// mailersend/lib/models/email/Identity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Identity = void 0;
class Identity {
    constructor(domainId, email, name, personalNote, replyToName, replyToEmail, addNote) {
        this.domain_id = domainId;
        this.email = email;
        this.name = name;
        this.personal_note = personalNote;
        this.reply_to_name = replyToName;
        this.reply_to_email = replyToEmail;
        this.add_note = addNote;
    }
    setDomainId(domainId) {
        this.domain_id = domainId;
        return this;
    }
    setEmail(email) {
        this.email = email;
        return this;
    }
    setName(name) {
        this.name = name;
        return this;
    }
    setPersonalNote(personalNote) {
        this.personal_note = personalNote;
        return this;
    }
    setReplyToName(replyToName) {
        this.reply_to_name = replyToName;
        return this;
    }
    setReplyToEmail(replyToEmail) {
        this.reply_to_email = replyToEmail;
        return this;
    }
    setAddNote(addNote) {
        this.add_note = addNote;
        return this;
    }
}
exports.Identity = Identity;
