// mailersend/lib/modules/email/Domain.module.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainModule = void 0;
const request_service_1 = require("../../services/request.service");
class DomainModule extends request_service_1.RequestService {
    constructor(apiKey, baseUrl) {
        super(apiKey, baseUrl);
    }
    create(domain) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.post(`/domains`, domain);
        });
    }
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/domains`, queryParams);
        });
    }
    single(domainId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/domains/${domainId}`);
        });
    }
    delete(domainId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.deleteReq(`/domains/${domainId}`);
        });
    }
    updateSettings(domainId, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.put(`/domains/${domainId}/settings`, data);
        });
    }
    recipients(domainId, queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/domains/${domainId}/recipients`, queryParams);
        });
    }
    dns(domainId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/domains/${domainId}/dns-records`);
        });
    }
    verify(domainId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/domains/${domainId}/verify`);
        });
    }
}
exports.DomainModule = DomainModule;
