// mailersend/lib/modules/email/Identity.module.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityModule = void 0;
const request_service_1 = require("../../services/request.service");
class IdentityModule extends request_service_1.RequestService {
    constructor(apiKey, baseUrl) {
        super(apiKey, baseUrl);
    }
    create(identity) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.post(`/identities`, identity);
        });
    }
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/identities`, queryParams);
        });
    }
    single(identityId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/identities/${identityId}`);
        });
    }
    singleByEmail(email) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.get(`/identities/email/${email}`);
        });
    }
    update(identityId, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.put(`/identities/${identityId}`, data);
        });
    }
    updateByEMail(email, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.put(`/identities/email/${email}`, data);
        });
    }
    delete(identityId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.deleteReq(`/identities/${identityId}`);
        });
    }
    deleteByEmail(email) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.deleteReq(`/identities/email/${email}`);
        });
    }
}
exports.IdentityModule = IdentityModule;
