// mailersend/lib/modules/SMS.module.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SMSModule = void 0;
const request_service_1 = require("../services/request.service");
const Activity_module_1 = require("./sms/Activity.module");
const Inbound_module_1 = require("./sms/Inbound.module");
const Message_module_1 = require("./sms/Message.module");
const Number_module_1 = require("./sms/Number.module");
const Recipient_module_1 = require("./sms/Recipient.module");
const Webhook_module_1 = require("./sms/Webhook.module");
class SMSModule extends request_service_1.RequestService {
    constructor(apiKey, baseUrl) {
        super(apiKey, baseUrl);
        this.activity = new Activity_module_1.SmsActivityModule(apiKey, baseUrl);
        this.number = new Number_module_1.SmsNumberModule(apiKey, baseUrl);
        this.message = new Message_module_1.SmsMessageModule(apiKey, baseUrl);
        this.inbound = new Inbound_module_1.SmsInboundModule(apiKey, baseUrl);
        this.recipient = new Recipient_module_1.SmsRecipientModule(apiKey, baseUrl);
        this.webhook = new Webhook_module_1.SmsWebhookModule(apiKey, baseUrl);
    }
    send(params) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.post("/sms", params);
        });
    }
}
exports.SMSModule = SMSModule;
